packages/drivers/odsp-driver/src/odspDelayLoadedDeltaStream.ts

```diff
@@ -220,9 +220,19 @@ export class OdspDelayLoadedDeltaStream {
 				this.currentConnection = connection;
 				return connection;
 			} catch (error) {
-				this.clearJoinSessionTimer();
-				this.cache.sessionJoinCache.remove(this.joinSessionKey);
-
+				// Remove join session information from cache only if it is an error is from socket event connect_document_error.
+				// Otherwise keep it in cache so that this session can be re-used after disconnection.
+				// Also keeping an undefined check here to account for any unknown code path that is unable to stamp the value as in that case also
+				// it is safer to clear join session cache and start over.
+				if (
+					// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+					(error as any).errorFrom === "connect_document_error" ||
```

Comment:
we should avoid casting, especially to any, whenever possible. in this case, creating a type that include the errorFrom property that can be used for both the error creator and here would reduce the chance of error. you could also add a typegaurd to inspect the error object to determine if it matches the type. generally, the typegaurd will do casting, but it pairs it with runtime checks to ensure the cast is valid and safe.

Reply:
@anthony-murphy Fixed now. Please re-review.